src/html-media-source.js

```diff
@@ -264,4 +264,48 @@ export default class HtmlMediaSource extends videojs.EventTarget {
     this.sourceBuffers.push(buffer);
     return buffer;
   }
+
+  /**
+   * add a property to the VSB indicating the current audio/video/text stream
+   * properties (sample rate, channels, etc) and fire an event
+   * when the info changes
+   *
+   * @private
+   * @param {String} type the type of stream to change info on (audio, video, text)
+   * @param {Object} info info about the currently muxed audio/video/text stream
+   */
+  changeInfo_(type, info) {
+    let propName = `${type}Info_`;
+
+    if (!this[propName]) {
+      this[propName] = info;
+      return;
+    }
+
+    let change = () => {
+      // copy current properties to another object
+      let old = videojs.mergeOptions({}, this[propName]);
+
+      this[propName] = info;
+      this.trigger({
+        revert: () => this[propName] = old,
```

Comment:
hmmm it might make more sense to do it like that. We trigger an audioinfo event and have HLS determine wether the info for the stream has changed, probably by doing something similar to this function, on every event.